### PR TITLE
Added sort for articles and topics filter

### PR DESCRIPTION
**Pull request by OJ423**
- Error handling is next on the to-do-list
- Created sorting for articles page
- Created sorting for topics page.

### CODE REVIEW

File: src/components/utils.js [hunk 1]
```diff
@@ -6,13 +6,24 @@ const instance = axios.create({
...
-    filterQuery = `?${Object.keys(filter)}=${Object.values(filter)}`
+// if (topic.topic !== undefined && sort.sort_by !== null ) {
+//   Query = `?${Object.keys(topic)}=${Object.values(topic)}&${Object.keys(sort)}=${Object.values(sort)}`
+// }
+// else if(sort.sort_by !== null && topic.topic === undefined) {
+//   Query = `?${Object.keys(sort)}=${Object.values(sort)}`
+// }
+// else if (sort.sort_by === null && topic.topic !== undefined) {
+//   Query = `?${Object.keys(topic)}=${Object.values(topic)}`
+// }
```
Comment (nc-bela-bertalan) [inline src/components/utils.js:17]: Please remove all commented-out code.

File: src/components/utils.js [hunk 2]
```diff
@@ -22,6 +33,25 @@ export function fetchArticles(filter = null) {
...
+export function fetchFilteredArticles(topic = null, sort = null) {
+  let Query = null
+  if (topic.topic !== undefined && sort.sort_by !== null ) {
+    Query = `?${Object.keys(topic)}=${Object.values(topic)}&${Object.keys(sort)}=${Object.values(sort)}`
+  }
+  else if(sort.sort_by !== null && topic.topic === undefined) {
+    Query = `?${Object.keys(sort)}=${Object.values(sort)}`
+  }
+  else if (sort.sort_by === null && topic.topic !== undefined) {
+    Query = `?${Object.keys(topic)}=${Object.values(topic)}`
```
Comment (nc-bela-bertalan) [inline src/components/utils.js:45]: Please look into the use of axios 'params' in the Notes and try to refactor this.

**Review by nc-bela-bertalan — COMMENTED** [review]
Hey Oliver, please see my other comments too. Also users should be able to sort articles by votes and comments as well and all of those be ordered either in ascending or descending order.

File: src/components/utils.js (hunk 1 above)
Comment (OJ423) [reply src/components/utils.js:17]: Done! And much better now.

**Pull request merged**